src/state/slices/postsSlice.ts

```diff
@@ -104,6 +106,15 @@ export const postsSlice = createSlice({
           : state.expandedPost.numberOfLikes--;
       }
     },
+    toggleFollow: (state, action: PayloadAction<boolean>) => {
+      return {
+        ...state,
+        expandedPost: {
+          ...state.expandedPost,
+          followStatus: !action.payload,
+        },
+      };
+    },
```

Comment:
Do you know if we have to return the entire state? Is it possible to mutate only the `expandedPost` object to keep it simple? Something like

```suggestion
    toggleFollow: (state, action: PayloadAction<boolean>) => {
      state.expandedPost.followStatus = !action.payload;
    },
```

Reply:
updated